File: backend/test_simple.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
DOBACK SOFT - PROCESADOR DE PRUEBA SIMPLE
===============================================================================
"""

import os
import sys
import json
import logging
from datetime import datetime
from typing import Dict, List, Optional, Tuple, Set
import psycopg2

# Configuración de logging
logging.basicConfig(
    level=logging.INFO,
    format='%(asctime)s - %(levelname)s - %(message)s',
    handlers=[
        logging.FileHandler('test_simple.log', encoding='utf-8'),
        logging.StreamHandler(sys.stdout)
    ]
)
logger = logging.getLogger(__name__)

# Configuración de directorios
DATA_DIR = os.path.join(os.path.dirname(__file__), 'data', 'datosDoback')

# Configuración de base de datos
DB_CONFIG = {
    'host': 'localhost',
    'port': 5432,
    'database': 'dobacksoft',
    'user': 'postgres',
    'password': 'postgres'
}

class SimpleTestProcessor:
    def __init__(self):
        self.data_dir = DATA_DIR
        self.db_config = DB_CONFIG
        self.organization_name = 'CMadrid'
        self.user_id = 'a8071944-989c-4627-b8a7-71aa03f24180'

    def run(self):
        """Ejecuta el procesador de prueba simple."""
        logger.info("🚀 Iniciando procesador de prueba simple")
        
        try:
            # Escanear archivos
            all_files = self._get_all_files()
            logger.info(f"📁 Total archivos encontrados: {len(all_files)}")
            
            # Agrupar por vehículo
            files_by_vehicle = {}
            for file_info in all_files:
                vehicle = file_info.get('vehicle', 'unknown')
                if vehicle not in files_by_vehicle:
                    files_by_vehicle[vehicle] = []
                files_by_vehicle[vehicle].append(file_info)
            
            # Mostrar resumen por vehículo
            logger.info("📊 === RESUMEN POR VEHÍCULO ===")
            for vehicle, files in files_by_vehicle.items():
                # Contar por tipo
                types = {}
                for file_info in files:
                    file_type = file_info.get('type')
                    types[file_type] = types.get(file_type, 0) + 1
                
                logger.info(f"  🚗 {vehicle}: {types}")
                
                # Verificar si tiene datos mínimos
                has_minimum = types.get('ESTABILIDAD', 0) > 0 or types.get('GPS', 0) > 0
                if has_minimum:
                    logger.info(f"     ✅ VÁLIDO - Tiene datos mínimos")
                else:
                    logger.info(f"     ❌ INVÁLIDO - No tiene datos mínimos")
            
            # Buscar vehículos válidos
            valid_vehicles = []
            for vehicle, files in files_by_vehicle.items():
                types = {}
                for file_info in files:
                    file_type = file_info.get('type')
                    types[file_type] = types.get(file_type, 0) + 1
                
                has_minimum = types.get('ESTABILIDAD', 0) > 0 or types.get('GPS', 0) > 0
                if has_minimum:
                    valid_vehicles.append(vehicle)
            
            logger.info(f"✅ Vehículos válidos encontrados: {len(valid_vehicles)}")
            logger.info(f"📋 Lista: {valid_vehicles}")
            
        except Exception as e:
            logger.error(f"Error en procesador de prueba: {e}")
            raise

    def _get_all_files(self) -> List[Dict]:
        """Obtiene todos los archivos del directorio de datos."""
        files = []
        
        for root, dirs, filenames in os.walk(self.data_dir):
            for filename in filenames:
                if filename.endswith(('.txt', '.csv')):
                    file_path = os.path.join(root, filename)
                    file_info = self._parse_filename(file_path)
                    if file_info:
                        files.append(file_info)
        
        return files

    def _parse_filename(self, file_path: str) -> Optional[Dict]:
        """Parsea el nombre del archivo para extraer información."""
        filename = os.path.basename(file_path)
        
        # Detectar tipo de archivo
        if 'CAN' in filename.upper():
            file_type = 'CAN'
        elif 'GPS' in filename.upper():
            file_type = 'GPS'
        elif 'ESTABILIDAD' in filename.upper():
            file_type = 'ESTABILIDAD'
        elif 'ROTATIVO' in filename.upper():
            file_type = 'ROTATIVO'
        else:
            return None
        
        # Extraer vehículo y fecha del nombre
        parts = filename.split('_')
        if len(parts) >= 3:
            vehicle = parts[1]
            date_str = parts[2]
            
            # Convertir fecha
            try:
                date = datetime.strptime(date_str, '%Y%m%d').date()
                return {
                    'path': file_path,
                    'filename': filename,
                    'type': file_type,
                    'vehicle': vehicle,
                    'date': date
                }
            except ValueError:
                return None
        
        return None

if __name__ == "__main__":
    processor = SimpleTestProcessor()
    processor.run()

File: backend/scripts/test-simple.ts
#!/usr/bin/env ts-node

import { PrismaClient } from '@prisma/client';
import { logger } from '../src/utils/logger';

const prisma = new PrismaClient();

async function testSimple() {
    try {
        logger.info('🧪 Iniciando prueba simple...');

        // 1. Verificar conexión a BD
        await prisma.$connect();
        logger.info('✅ Conexión a BD exitosa');

        // 2. Crear organización si no existe
        let org = await prisma.organization.findFirst({
            where: { id: 'CMadrid' }
        });

        if (!org) {
            org = await prisma.organization.create({
                data: {
                    id: 'CMadrid',
                    name: 'CMadrid',
                    apiKey: 'cmadrid-api-key-2025'
                }
            });
            logger.info('✅ Organización creada');
        } else {
            logger.info('ℹ️ Organización ya existe');
        }

        // 3. Crear vehículo de prueba
        let vehicle = await prisma.vehicle.findFirst({
            where: { name: 'DOBACK022' }
        });

        if (!vehicle) {
            vehicle = await prisma.vehicle.create({
                data: {
                    name: 'DOBACK022',
                    licensePlate: 'DOBACK022',
                    organizationId: 'CMadrid',
                    status: 'ACTIVE',
                    model: 'DOBACK',
                    identifier: 'DOBACK022',
                    type: 'VAN'
                }
            });
            logger.info('✅ Vehículo creado');
        } else {
            logger.info('ℹ️ Vehículo ya existe');
        }

        // 4. Listar archivos disponibles
        const fs = require('fs');
        const path = require('path');

        const dataPath = path.join(__dirname, '../data/datosDoback/CMadrid');

        if (fs.existsSync(dataPath)) {
            const vehicles = fs.readdirSync(dataPath, { withFileTypes: true })
                .filter(dirent => dirent.isDirectory())
                .map(dirent => dirent.name);

            logger.info(`📁 Vehículos encontrados: ${vehicles.join(', ')}`);

            // Contar archivos por tipo
            let totalFiles = 0;
            const fileTypes = { GPS: 0, CAN: 0, ESTABILIDAD: 0, ROTATIVO: 0 };

            for (const vehicleDir of vehicles) {
                const vehiclePath = path.join(dataPath, vehicleDir);
                for (const dataType of ['GPS', 'CAN', 'estabilidad', 'rotativo']) {
                    const typePath = path.join(vehiclePath, dataType);
                    if (fs.existsSync(typePath)) {
                        const files = fs.readdirSync(typePath)
                            .filter(file => file.endsWith('.txt'));

                        const typeKey = dataType.toUpperCase() as keyof typeof fileTypes;
                        fileTypes[typeKey] += files.length;
                        totalFiles += files.length;
                    }
                }
            }

            logger.info('📊 Resumen de archivos:');
            logger.info(`  - GPS: ${fileTypes.GPS} archivos`);
            logger.info(`  - CAN: ${fileTypes.CAN} archivos`);
            logger.info(`  - ESTABILIDAD: ${fileTypes.ESTABILIDAD} archivos`);
            logger.info(`  - ROTATIVO: ${fileTypes.ROTATIVO} archivos`);
            logger.info(`  - Total: ${totalFiles} archivos`);

        } else {
            logger.error(`❌ Directorio no encontrado: ${dataPath}`);
        }

        logger.info('✅ Prueba simple completada');

    } catch (error) {
        logger.error('❌ Error en prueba simple:', error);
        throw error;
    } finally {
        await prisma.$disconnect();
    }
}

// Ejecutar si es llamado directamente
if (require.main === module) {
    testSimple().catch((error) => {
        logger.error('❌ Error fatal:', error);
        process.exit(1);
    });
}

export { testSimple };
